src/v/cluster/CMakeLists.txt

```diff
@@ -241,6 +241,7 @@ v_cc_library(
     data_migration_types.cc
     data_migration_table.cc
     data_migration_frontend.cc
+    data_migration_worker.cc
```

Comment:
please make the commit title meaningful and provide more context in commit message about the place of the worker in data migrations subsystem